cold-wallet/seed/secure-seed.ts

```diff
@@ -1,15 +1,39 @@
 import Config from '../../shared/config';
-import { Logger } from '../../shared/logger';
+import { Shamir } from '../../shared/shamir';
+import { Util } from '../../shared/util';
 
 export class SecureSeed {
-  private static logger = new Logger('Secure Seed');
-
   static async splitAndStore(seed: string[]): Promise<void> {
-    this.logger.info('"safely" secured following seed :D');
-    this.logger.info('', seed);
+    // generate shamir shares
+    const secret = seed.join(' ');
+    const shares = Shamir.split(secret, Config.wallet.seed.shareCount, Config.wallet.seed.thresholdCount);
+
+    // write to file
+    shares.forEach((share, i) => Util.writeFileRaw(Config.wallet.seed.fileName(i + 1), share));
   }
 
   static async read(): Promise<string[]> {
-    return Config.wallet.seed?.split(',') ?? [];
+    // read the shares from file
+    const shares = [];
+    for (let i = 0; i < Config.wallet.seed.shareCount && shares.length < Config.wallet.seed.thresholdCount; i++) {
+      const share = this.readShamirShare(i);
```

Comment:
here we start with 0. am i missing something?

Reply:
Hm, good catch.